base/uk.ac.stfc.isis.ibex.ui.banner/src/uk/ac/stfc/isis/ibex/ui/banner/views/BannerView.java

```diff
@@ -141,6 +144,11 @@ private Collection<CustomControlModel> convertBannerButtons(Collection<BannerBut
      */
     private void setBanner(final Collection<BannerItemModel> indicatorModels, final Collection<CustomControlModel> controlModels) {
         disposeBanner();
+        
+        // Ensure that previous models get closed.
+        activeModels.forEach(CustomControlModel::close);
```

Comment:
Unless I'm missing something, it doesn't seem like anything ever gets added to this collection?

Reply:
You're right, nothing gets added, but the entire collection gets reassigned.

So the first time it creates the banner it will loop over nothing (and therefore do nothing), but then the collection gets reassigned to the new set of `controlModels`. So then the next time the banner is set, there are now the *old* set of models in the collection, which get closed just before being reassigned again next time `setBanner` is called.

Reply:
Ah, I see. That was lazy reading on my part. Thanks